docs/machine.md

```diff
@@ -0,0 +1,30 @@
+# Using with Docker Toolbox and Machine
+
+In development, we recommend using [Docker Toolbox](https://www.docker.com/products/docker-toolbox) to set up Docker. It includes a tool called Machine which will create a VM running Docker Engine and point your shell at it using environment variables.
```

Comment:
I dislike the phrasing "we recommend using Docker Toolbox", because while I understand it's the product we at Docker want people to use, I feel it's irrelevant to docker-py in particular.

Can we keep a more neutral approach here?